library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -0,0 +1,276 @@
+import com.github.javaparser.utils.ProjectRoot;
+import com.github.javaparser.utils.SourceRoot;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utilized for generating an astubx file from a directory containing annotated Java source code.
+ *
+ * <p>This class utilizes com.github.javaparser APIs to analyze Java source files within a specified
+ * directory. It processes the annotated Java source code to generate an astubx file that contains
+ * the required annotation information to be able to generate library models.
+ */
+public class LibraryModelGenerator {
+
+  public void generateAstubxForLibraryModels(String inputSourceDirectory, String outputDirectory) {
+    Map<String, MethodAnnotationsRecord> methodRecords = processDirectory(inputSourceDirectory);
+    writeToAstubx(outputDirectory, methodRecords);
+  }
+
+  /**
+   * Parses all the source files within the directory using javaparser.
+   *
+   * @param sourceDirectoryRoot Directory containing annotated java source files.
+   * @return a Map containing the Nullability annotation information from the source files.
+   */
+  private Map<String, MethodAnnotationsRecord> processDirectory(String sourceDirectoryRoot) {
+    Map<String, MethodAnnotationsRecord> methodRecords = new LinkedHashMap<>();
+    Path root = dirnameToPath(sourceDirectoryRoot);
+    AnnotationCollectorCallback ac = new AnnotationCollectorCallback(methodRecords);
+    CollectionStrategy strategy = new ParserCollectionStrategy();
+    // Required to include directories that contain a module-info.java, which don't parse by
+    // default.
+    strategy.getParserConfiguration().setLanguageLevel(LanguageLevel.JAVA_17);
+    ProjectRoot projectRoot = strategy.collect(root);
+
+    projectRoot
+        .getSourceRoots()
+        .forEach(
+            sourceRoot -> {
+              try {
+                sourceRoot.parse("", ac);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    return methodRecords;
+  }
+
+  /**
+   * Writes the Nullability annotation information into the output directory as an astubx file.
+   *
+   * @param outputPath Output Directory.
+   * @param methodRecords Map containing the collected Nullability annotation information.
+   */
+  private void writeToAstubx(
+      String outputPath, Map<String, MethodAnnotationsRecord> methodRecords) {
+    if (methodRecords.isEmpty()) {
+      return;
+    }
+    Map<String, String> importedAnnotations =
+        ImmutableMap.of(
+            "Nonnull", "javax.annotation.Nonnull",
+            "Nullable", "javax.annotation.Nullable");
```

Comment:
Why are we using the `javax` annotations here and not JSpecify? Some limitation with `StubxWriter`?

Reply:
https://github.com/uber/NullAway/blob/3bde26cdad1c06f0312cd85c22244c9ad5afea2f/nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java#L221

No limitation with `StubxWriter` , just used it because the check here considers the `javax` annotation and I probably didn't consider adding an OR and including the jspecify annotation in the check at the time.